mockall/src/lib.rs

```diff
@@ -1656,3 +1659,85 @@ impl Sequence {
         handle
     }
 }
+
+/// Used to capture values passed to mocked methods, so you can assert on them later.
+///
+/// # Examples
+/// ```
+/// # use std::ops::Range;
+/// use mockall::*;
+///
+/// #[automock]
+/// trait Foo {
+///     fn foo(&self, some_param: i32);
+/// }
+///
+/// let mut mock = MockFoo::new();
+/// let int_captor = captor();
+///
+/// mock.expect_foo()
+///     .with(int_captor.capture_values())
+///     .returning(|_| ());
+///
+/// mock.foo(41);
+/// mock.foo(42);
+///
+/// assert_eq!(int_captor.values().as_slice(), &[41, 42]);
+/// assert_eq!(int_captor.last_value(), Some(42));
+///
+/// int_captor.reset();
+///
+/// assert_eq!(int_captor.values().len(), 0);
+/// ```
+///
+pub struct Captor<T> {
+    values: Arc<Mutex<Vec<T>>>,
+}
+
+/// Create a new captor. See the documentation for [Captor](Captor)
+pub fn captor<T>() -> Captor<T> {
+    Captor { values: Arc::new(Mutex::new(vec![])) }
+}
+
+impl<T: Clone> Captor<T> {
+    /// Retrieve the list of all captured values, in chronological order
+    pub fn values(&self) -> Vec<T> {
+        self.lock().clone()
+    }
+
+    /// Retrieve the last captured value, if any.
+    pub fn last_value(&self) -> Option<T> {
+        self.lock().last().cloned()
+    }
+
+    /// Forget all previously captured values. Reset this `Captor` to its initial state.
+    pub fn reset(&self) {
+        self.lock().clear()
+    }
+
+    /// Start capturing values. See the documentation for [Captor](Captor)
+    pub fn capture_values(&self) -> impl Predicate<T> {
```

Comment:
Just in the interest of reducing typing, I suggest shortening the name:
```suggestion
    pub fn capture(&self) -> impl Predicate<T> {
```